OTAnalytics/domain/observer.py

```diff
@@ -0,0 +1,23 @@
+from abc import ABC
+from typing import Generic, TypeVar
+
+OBSERVER = TypeVar("OBSERVER")
+
+
+class Registrable(ABC, Generic[OBSERVER]):
```

Comment:
We might add a second generic to reduce duplication of `...Subject` classes further :-) See comment in `section.py`.